components/DirectoryListing.tsx

```diff
@@ -84,83 +93,151 @@ export function DirectoryListing(props: DirectoryListingProps) {
-                            }`}
-                          </span>
-                        </div>
-                      </label>
-                    </td>
-                  </tr>
+                  <a href={props.repositoryURL} class={tw`link ml-4`}>
+                    Repository
+                  </a>
                 )}
-              {hiddenItems.map((entry: Entry, i: number) => {
-                const isLastItem = hiddenItems.length - 1 === i;
-                return (
-                  <TableRow
-                    key={i}
-                    entry={entry}
-                    href={buildEntryURL(props.path, entry)}
-                    isLastItem={isLastItem}
-                    isHiddenItem={true}
+            </div>
+            <div
+              class={tw`inline-block ml-4 inline-flex shadow-sm rounded-md`}
+            >
+              <a
+                href={searchDoc.href}
+                class={tw
+                  `relative inline-flex items-center px-1.5 py-1.5 rounded-l-md border border-gray-300 text-sm font-medium text-gray-500 hover:bg-gray-50 ${
+                    !dirview ? "bg-white" : "bg-gray-100"
+                  }`}
+              >
+                <span class={tw`sr-only`}>Documentation</span>
+                <svg
+                  xmlns="http://www.w3.org/2000/svg"
+                  class="h-5 w-5"
+                  viewBox="0 0 20 20"
+                  fill="currentColor"
+                >
+                  <path d="M9 4.804A7.968 7.968 0 005.5 4c-1.255 0-2.443.29-3.5.804v10A7.969 7.969 0 015.5 14c1.669 0 3.218.51 4.5 1.385A7.962 7.962 0 0114.5 14c1.255 0 2.443.29 3.5.804v-10A7.968 7.968 0 0014.5 4c-1.255 0-2.443.29-3.5.804V12a1 1 0 11-2 0V4.804z" />
+                </svg>
+              </a>
+              <a
+                href={searchDir.href}
+                class={tw
+                  `-ml-px relative inline-flex items-center px-1.5 py-1.5 rounded-r-md border border-gray-300 text-sm font-medium text-gray-500 hover:bg-gray-50 ${
+                    dirview ? "bg-white" : "bg-gray-100"
+                  }`}
+              >
+                <span class={tw`sr-only`}>Directory Listing</span>
+                <svg
+                  xmlns="http://www.w3.org/2000/svg"
+                  class="h-5 w-5"
+                  viewBox="0 0 20 20"
+                  fill="currentColor"
+                >
+                  <path
+                    fill-rule="evenodd"
+                    d="M12.316 3.051a1 1 0 01.633 1.265l-4 12a1 1 0 11-1.898-.632l4-12a1 1 0 011.265-.633zM5.707 6.293a1 1 0 010 1.414L3.414 10l2.293 2.293a1 1 0 11-1.414 1.414l-3-3a1 1 0 010-1.414l3-3a1 1 0 011.414 0zm8.586 0a1 1 0 011.414 0l3 3a1 1 0 010 1.414l-3 3a1 1 0 11-1.414-1.414L16.586 10l-2.293-2.293a1 1 0 010-1.414z"
+                    clip-rule="evenodd"
                   />
-                );
-              })}
-            </tbody>
-          </table>
+                </svg>
```

Comment:
For the various SVGs, can you move them into their own components?

Reply:
Done; moved most, some are still left, but are a bit more complicated and dont want to keep this simple